src/README.md

```diff
@@ -1,28 +1,98 @@
 
+The project requires python3.9 +, we will go through the installation of pyenv to easily switch and virtual environment 
 
-### Config file
-The config file is the most important file when running the code. Always be sure of what config you are using. Our `main.py` file uses a combination of `algo_config.py` and `sys_config.py` by default. These files can have multiple configurations defined within them and the provision to select one as default. 
+## Setting Up `pyenv`
 
-We have intentionally kept configuration files as a python file which is typically a big red flag in software engineering. But we did this because it enables plenty of quick automations and flexibility. Be very careful with the config file because it is easy to overlook some of the configurations such as device ids, number of clients etc.
+`pyenv` allows you to easily switch between multiple versions of Python. Here's how to set it up:
 
-### Reproducibility
-One of the awesome things about this project is that whenever you run an experiment, all the source code, logs, and model weights are saved in a separate folder. This is done to ensure that you can reproduce the results by looking at the code that was responsible for the results. The naming of the folder is based on the keys inside the config file. That also means you can not run the same experiment again without renaming/deleting the previous experimental run. The code automatically asks you to press `r` to remove and create a new folder. Be careful you are not overwriting someone else's results.
+### 1. Install pyenv 
 
-### Logging
-We log the results in the console and also in a log file that captures the same information. We also log a few metrics for the tensorboard. 
+```bash
+curl https://pyenv.run | bash
+```
+
+## Add pyenv to Your Shell:
+Add the following lines to your shell configuration file (~/.bashrc, ~/.zshrc, etc.):
+
+```bash
+export PATH="$HOME/.pyenv/bin:$PATH"
+eval "$(pyenv init --path)"
+eval "$(pyenv init -)"
+eval "$(pyenv virtualenv-init -)"
+
+```
+
+## Install Python Version
+
+Use pyenv to install Python 3.9 or later:
+
+```bash
+pyenv install 3.9.10
+pyenv global 3.9.10
+
+```
+
+## Verify Installation
+python --version
+
+```bash
+python --version
+
+```
+
+## Setting Up the Environment
+Create and activate a virtual environment using pyenv and pyenv-virtualenv:
+
+### Install pyenv-virtualenv:
+
+   #### On Linux:
+
+```bash
+git clone https://github.com/pyenv/pyenv-virtualenv.git ~/.pyenv/plugins/pyenv-virtualenv
+```
+ #### On Windows:
 
-The tensorboard logs can be viewed by running tensorboard as follows:<br>
-`tensorboard --logdir=expt_dump/ --host 0.0.0.0`<br>
-Assuming `expt_dump` is the folder where the experiment logs are stored.
+```bash 
+git clone https://github.com/pyenv-win/pyenv-win.git %USERPROFILE%\.pyenv
+```
 
-After a successful run with 50 epochs, the Tensorboard experiment log should look something like below:
```

Comment:
The changes in this readme removes these lines, plus the sample plot. Kindly recheck and update the same, thanks!

Reply:
Resolved!